working/control-flow-collections/feature-specification.md

```diff
@@ -684,10 +684,137 @@ shows that a variable has some type and promotion isn't otherwise aborted.
 
 ### Const collections
 
-An `if` element can be used in a const collection, provided its condition
-expression and body elements are all const.
+We have to be careful to ensure that arbitrary computation doesn't happen due
+to `if` or `for` appearing in a constant collection. To that end, we only allow an element in a const collection when:
 
-`for` elements cannot be used in const lists and maps.
+*   It is an expression element and the expression is constant.
+
+*   It is an `if` element and the condition expression is constant and both body
+    elements are allowed.
+
+*   It is a `for` element with an `in` clause. The iterator expression must be
+    a constant expression. In a constant list literal, the expression must
+    evaluate to an object created by a constant list or set literal, as in:
+
+    ```dart
+    const list = [1, 2];
+    const set = {3, 4};
+    const result = const [
+      for (var i in list) i,
+      for (var i in set) i
+    ];
+    ```
+
+    This restriction ensures that iterating over the iterable does not call
+    user code.
+
```

Comment:
The spread proposal covers that, yes. I'm not planning to directly merge the two proposal docs, but I'll try to make sure this one calls out to spread where appropriate.

I can merge them if you really want, but I think it will just add to confusion now and be a lot of busywork.